src/VscPreviewExtension/src/extension.ts

```diff
@@ -0,0 +1,270 @@
+                    provider.update(documentUri);
+                }
+                else {
+                    // The first one and the result is truncated
+                    previewResult = tmp;
+                    multipleRead = true;
+                }
+            }
+            else {
+                if (endCharCode != ENDCODE) {
+                    // The result is truncated and this is not the last one
+                    previewResult += tmp;
+                }
+                else {
+                    // The result is truncated and this is the last one
+                    previewResult += tmp;
+                    multipleRead = false;
+                    provider.update(documentUri);
+                }
+            }
+        });
+
+        this._spawn.stderr.on('data', function (data) {
+            console.log("error " + data + '\n');
+        });
+
+        this._spawn.on('exit', function (code) {
+            console.log('child process exit with code ' + code);
+        });
+    }
+
+    private sendtext() {
+        let editor = window.activeTextEditor;
+        if (!editor) {
+            return;
+        }
+        let doc = editor.document;
+        let docContent = doc.getText();
+        let fileName = doc.fileName;
+        let rtPath = workspace.rootPath;
+        let filePath;
+        if (!rtPath) {
+            let indexOfFilename = fileName.lastIndexOf('\\');
+            rtPath = fileName.substr(indexOfFilename - 1);
+            filePath = fileName.substring(0, indexOfFilename);
+        }
+        else {
+            let rtpath_length = rtPath.length;
+            filePath = fileName.substr(rtpath_length + 1, fileName.length - rtpath_length);
+        }
+        if (doc.languageId === "markdown") {
+            let numOfRow = doc.lineCount;
+            this._spawn.stdin.write(rtPath + '\n');
+            this._spawn.stdin.write(filePath + '\n');
+            this._spawn.stdin.write(numOfRow + '\n');
+            this._spawn.stdin.write(docContent + '\n');
+        }
+    }
+
+    public callDfm() {
+        if (this._isFirstTime) {
+            // In the firt time, if wait for the timeout, activeTextEditor will be translate to the preview window.
+            this._isFirstTime = false;
+            this.sendtext();
+        }
+        else if (!this._waiting) {
+            this._waiting = true;
+            setTimeout(() => {
+                this._waiting = false;
+                this.sendtext();
+            }, 300);
+        }
+    }
+}
+
+class MDDocumentContentProvider implements TextDocumentContentProvider {
+    private _context: ExtensionContext;
+    private _onDidChange = new EventEmitter<Uri>();
+    private _waiting: boolean;
+
+    constructor(context: ExtensionContext) {
+        this._context = context;
+        this._waiting = false;
+    }
+
+    private getMediaPath(mediaFile): string {
+        return this._context.asAbsolutePath(path.join('media', mediaFile));
```

Comment:
the default markdown also use 'media' but not  'style' ,be 
Consisten better?

---
In reply to: [75425446](https://github.com/dotnet/docfx/pull/616#discussion_r75425446) [](ancestors = 75425446)